### PR TITLE
add lecture 09: Data formats

### PR DESCRIPTION
**Pull request by riederd**

### CODE REVIEW

File: 09_data_formats_python/09_data_formats_python.Rmd [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+## Data formats
+
+
+NumPy arrays are different from Pandas DataFrame:
+
+  - NumPy stores data as one contiguous block
+  - the whole NumPy array must have one data type
+  - Pandas DataFrames store data as individual columns
+  - the DataFrame my have columns of different data types
```
Comment (grst) [inline 09_data_formats_python/09_data_formats_python.Rmd]: might be worth mentioning that a pandas dataframe consists of arrays. Every column is a numpy array and can be accessed using `df[col].values`.

File: 09_data_formats_python/09_data_formats_python.Rmd [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+  - if a certain format is used by many others in my field, maybe there is reasons for using it.
+  - is the file format good for my data format (is it fast/space efficient/easy to use)?
+  - do I need a human-readable format or is it ok to work on it using programming languages?
+  - do I want to archive / share the data or do I just want to store it while I’m working?
+
+
+
+## Data fromats: CSV/TSV
+
+    - Type: Text format
```
Comment (grst) [inline 09_data_formats_python/09_data_formats_python.Rmd:116]: Disadvantages:
 * doesn't preserve data types
 * Not standardized (there are many "dialects")

File: 09_data_formats_python/09_data_formats_python.Rmd [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+```
+</div>
+
+
+## Data formats: JSON
+
+    - Type: Text format
+    - Packages needed: None (json-module is included with Python).
+    - Space efficiency: Ok.
+    - Good for sharing/archival: No! See warning below.
```
Comment (grst) [inline 09_data_formats_python/09_data_formats_python.Rmd]: I'm not sure I would agree with that for the following reasons: 

advantages
 * standardized (ISO standard, there's only one correct way of writing JSON)
 * preserves data types

A fair point is, though that if no specification it's given it's a pain to figure out what is what, way more than with a tabular format.

File: 09_data_formats_python/09_data_formats_python.Rmd [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+    - Space efficiency: Bad.
+    - Good for sharing/archival: Maybe.
+    - Tidy data:
+        - Speed: Bad
+        - Ease of use: Good
+    - Array data:
+        - Speed: Bad
+        - Ease of use: Ok
+    - Best use cases: Sharing data in many fields. Quick data analysis.
+
```
Comment (grst) [inline 09_data_formats_python/09_data_formats_python.Rmd:282]: Compulsory link: 
https://www.kristianbrock.com/post/send-me-data/
(how to make excel sheets less painful for bioinformaticians)

File: 09_data_formats_python/09_data_formats_python.Rmd [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+        - Ease of use: Good
+    - Array data:
+        - Speed: -
+        - Ease of use: -
+    - Best use cases: Temporary storage of tidy data.
+
+Feather is a file format for storing data frames quickly. There are libraries for Python, R and Julia
+
+
+## Data fromats: Feather
```
Comment (grst) [inline 09_data_formats_python/09_data_formats_python.Rmd]:
```suggestion
## Data fromats: Parquet
```

File: 09_data_formats_python/09_data_formats_python.Rmd [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+    - Packages needed: openpyxl
+    - Space efficiency: Bad.
+    - Good for sharing/archival: Maybe.
+    - Tidy data:
+        - Speed: Bad
+        - Ease of use: Good
+    - Array data:
+        - Speed: Bad
+        - Ease of use: Ok
+    - Best use cases: Sharing data in many fields. Quick data analysis.
```
Comment (grst) [inline 09_data_formats_python/09_data_formats_python.Rmd:280]: Best use case: manual data entry

File: 09_data_formats_python/09_data_formats_python.Rmd (hunk 3 above)
Comment (riederd) [reply 09_data_formats_python/09_data_formats_python.Rmd]: You're right

File: 09_data_formats_python/09_data_formats_python.Rmd (hunk 4 above)
Comment (riederd) [reply 09_data_formats_python/09_data_formats_python.Rmd:282]: cool link!

File: 09_data_formats_python/09_data_formats_python.Rmd (hunk 5 above)
Comment (riederd) [reply 09_data_formats_python/09_data_formats_python.Rmd]: good catch

File: 09_data_formats_python/09_data_formats_python.Rmd (hunk 6 above)
Comment (riederd) [reply 09_data_formats_python/09_data_formats_python.Rmd:280]: Added

**Pull request merged**